### PR TITLE
Reader: Convert PostTime to an es6 class

### PR DESCRIPTION
**Pull request by blowery**
No functional changes, just moving to es6.

To test, pull up the main reader stream and wait a few minutes. The humane dates on the posts should change over time.

### CODE REVIEW

File: client/reader/post-time/index.jsx [hunk 1]
```diff
@@ -1,54 +1,49 @@
...
+	}
 
-	_update: function( date ) {
+	_update = date => {
 		date = date || this.props.date;
-		this.smartSetState( { ago: humanDate( date ) } );
-	},
+		this.smartSetState( {
+			ago: humanDate( date ),
+			full: moment( date ).format( 'llll' )
```
Comment (omarjackman) [inline client/reader/post-time/index.jsx]: Maybe we could make this name more descriptive, maybe `fullDate`, `longDate`, `formatedDate` etc

File: client/reader/post-time/index.jsx [hunk 2]
```diff
@@ -1,54 +1,49 @@
...
-	componentWillUnmount: function() {
+	componentWillUnmount() {
 		ticker.off( 'tick', this._update );
-	},
-
-	smartSetState: smartSetState,
+	}
 
-	_update: function( date ) {
+	_update = date => {
```
Comment (bluefuton) [inline client/reader/post-time/index.jsx]: Can we drop the underscore on the method name?

[Coding guidelines](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md) say:

> Avoid prefixing method names with _.

File: client/reader/post-time/index.jsx (hunk 1 above)
Comment (blowery) [inline client/reader/post-time/index.jsx]: done

File: client/reader/post-time/index.jsx [hunk 3]
```diff
@@ -1,54 +1,49 @@
 /**
  * External dependencies
  */
-var debug = require( 'debug' )( 'calyso:reader:post-time' ),
-	React = require( 'react' ),
-	PureRenderMixin = require( 'react-pure-render/mixin' );
-
+import React from 'react';
+import { PureComponent } from 'react';
```
Comment (aduth) [inline client/reader/post-time/index.jsx]: One-liner:

```js
import React, { PureComponent } from 'react';
```

**Comment by aduth** [comment]
What's the point of `smartSetState` when used in combination with a pure component, since the pure logic will avoid the re-render if state hasn't changed. With both, there would be two shallow comparisons run on each changing state.

Aside: The implementation of `smartSetState` could be simplified using `react-addons-shallow-compare`

**Comment by blowery** [comment]
It avoids enquing a reconciliation pass, most of the time. This component would be calling setState on every tick, which isn't strictly necessary...

**Pull request merged**